REQUIRE

```diff
@@ -1,5 +1,5 @@
 julia 0.5
-DataArrays 0.3.4
+DataArrays 0.5
```

Comment:
If we remove the `describe` code here then we need DataArrays 0.5, otherwise `describe` won't be able to handle NAs.

What if instead of removing the `describe` code currently in DataFrames outright, use an `include_string` macro to only use it when the user isn't on Julia v0.6+ and DataArrays v0.5+? e.g.
```julia
if Pkg.installed("DataArrays") >= v"0.5"
    @include_string("""
                    ... current describe code for DataArrays in DataFrames ...
                   )
else
    do nothing, since DataArrays >= v"0.5" has the new describe code
end
```

I can add version specific tests against the current style of describe in DataFrames and the udpated style in DataArrays v0.5+, and I think everything else can remain untouched.